src/pages/Product/index.vue

```diff
@@ -1,15 +1,36 @@
 <template>
   <back-button />
-  <product-layout :product="product" />
+  <product-layout
+    v-if="product"
+    @on-add-to-shopping-cart="toggleProduct(product as Product)"
+    :is-product-in-cart="checkIsProductInCart(product.id, shoppingCart)"
+    :product="product"
+  />
 </template>
 
 <script setup lang="ts">
 import ProductLayout from './Product.layout.vue';
-import BackButton from '../../components/BackButton';
-import { handleProductQuery } from './utils/handleProductQuery.ts';
+import BackButton from 'src/components/BackButton';
+import { handleProductQuery } from './utils/handleProductQuery';
+import {
+  handleShoppingCart,
+  checkIsProductInCart,
+  confirmBeforeExit,
+  Product,
+} from 'src/utils';
+import { useRoute } from 'vue-router';
 import './styles.scss';
 
+const route = useRoute();
+
 const { product, productId, getProductDetail } = handleProductQuery();
+const { shoppingCart, toggleProduct } = handleShoppingCart();
+
+confirmBeforeExit({
+  currentRouteParent: route.matched[0].path,
```

Comment:
Please add a guard in the rare case that matched is empty

Reply:
I think is a rare case hehe, but will apply just in case. Thanks for the suggestion